src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs

```diff
@@ -50,8 +64,11 @@ public static IServiceCollection Configure<TOptions>(this IServiceCollection ser
                 throw new ArgumentNullException(nameof(configureOptions));
             }
 
-            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureOptions<TOptions>(configureOptions));
+            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureNamedOptions<TOptions>(name, configureOptions));
             return services;
         }
+
+        public static IServiceCollection ConfigureAll<TOptions>(this IServiceCollection services, Action<TOptions> configureOptions) where TOptions : class
+            => services.Configure(name: null, configureOptions: configureOptions);
```

Comment:
You'd basically be unable to use existing code today which has Configures with named options since every instance would pickup the default configuration.